api/proto/teleport/legacy/client/proto/authservice.proto

```diff
@@ -209,6 +209,12 @@ message OpenSSHCertRequest {
   ];
   // Cluster is the Teleport cluster name.
   string Cluster = 4 [(gogoproto.jsontag) = "cluster"];
+  repeated string Roles = 5 [(gogoproto.jsontag) = "roles"];
```

Comment:
Can we just pass the user's TLS certificate instead? This way, the Auth server can parse the user's certificate into a `tlsca.Identity` and use this for the normal certificate issuing flow - https://github.com/gravitational/teleport/blob/06a3864fc33e8db970afb82d1a17f6b53856121a/lib/auth/auth_with_roles.go#L2669. This should circumvent any similar issues with impersonation, access request roles, etc.